### ISSUE TITLE
[BUG] CRC often unreachable on Windows after suspend

### ISSUE THREAD

**Issue by mltucker** [crc-org/crc#3394]
Any help in figuring out why my CRC installation is unstable would be highly appreciated! Thanks :).

### General information

  * OS: Windows
  * Hypervisor: Hyper-V
  * Did you run `crc setup` before starting it (Yes/No)? Yes
  * Running CRC on: PC

## CRC version
```bash
# crc version
WARN A new version (2.10.1) has been published on https://developers.redhat.com/content-gateway/file/pub/openshift-v4/clients/crc/2.10.1/crc-windows-installer.zip
CRC version: 2.9.0+589ab2cd
OpenShift version: 4.11.3
Podman version: 4.2.0
```
  
## CRC status
```bash
# crc status --log-level debug
DEBU CRC version: 2.9.0+589ab2cd
DEBU OpenShift version: 4.11.3
DEBU Podman version: 4.2.0
DEBU Running 'crc status'
DEBU Checking file: C:\Users\m\.crc\machines\crc\.crc-exist
DEBU Checking file: C:\Users\m\.crc\machines\crc\.crc-exist
DEBU Running 'Hyper-V\Get-VM crc | Select-Object -ExpandProperty State'
DEBU Running SSH command: df -B1 --output=size,used,target /sysroot | tail -1
DEBU Using ssh private keys: [C:\Users\m\.crc\machines\crc\id_ecdsa C:\Users\m\.crc\cache\crc_hyperv_4.11.3_amd64\id_ecdsa_crc]
DEBU SSH command results: err: dial tcp 127.0.0.1:2222: connectex: No connection could be made because the target machine actively refused it., output:
DEBU Cannot get root partition usage: ssh command error:
command : df -B1 --output=size,used,target /sysroot | tail -1
err     : dial tcp 127.0.0.1:2222: connectex: No connection could be made because the target machine actively refused it.
DEBU cannot get OpenShift status: Get "https://api.crc.testing:6443/apis/config.openshift.io/v1/clusteroperators": dial tcp 127.0.0.1:6443: connectex: No connection could be made because the target machine actively refused it.
CRC VM:          Running
OpenShift:       Unreachable (v4.11.3)
Podman:
Disk Usage:      0B of 0B (Inside the CRC VM)
Cache Usage:     36.73GB
Cache Directory: C:\Users\m\.crc\cache
```

## CRC config
```bash
# crc config view
- consent-telemetry                     : no
```

## Host Operating System
```bash
# systeminfo
Host Name:                 PC-2018
OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.19043 N/A Build 19043
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          Windows User
Registered Organization:
Product ID:                00330-80000-00000-AA290
Original Install Date:     11/9/2020, 10:32:19 PM
System Boot Time:          10/12/2022, 3:44:12 AM
System Manufacturer:       System manufacturer
System Model:              System Product Name
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 158 Stepping 10 GenuineIntel ~3696 Mhz
BIOS Version:              American Megatrends Inc. 0802, 11/2/2017
Windows Directory:         C:\WINDOWS
System Directory:          C:\WINDOWS\system32
Boot Device:               \Device\HarddiskVolume6
System Locale:             en-gb;English (United Kingdom)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC+01:00) Amsterdam, Berlin, Bern, Rome, Stockholm, Vienna
Total Physical Memory:     65,459 MB
Available Physical Memory: 39,179 MB
Virtual Memory: Max Size:  75,187 MB
Virtual Memory: Available: 38,298 MB
Virtual Memory: In Use:    36,889 MB
Page File Location(s):     D:\pagefile.sys
Domain:                    WORKGROUP
Logon Server:              \\PC-2018
Hotfix(s):                 19 Hotfix(s) Installed.
                           [01]: KB5017262
                           [02]: KB4577266
                           [03]: KB4577586
                           [04]: KB4580325
                           [05]: KB4586864
                           [06]: KB4589212
                           [07]: KB4593175
                           [08]: KB4598481
                           [09]: KB5000736
                           [10]: KB5012170
                           [11]: KB5018410
                           [12]: KB5006753
                           [13]: KB5007273
                           [14]: KB5011352
                           [15]: KB5014035
                           [16]: KB5014671
                           [17]: KB5015895
                           [18]: KB5016705
                           [19]: KB5005699
Network Card(s):           6 NIC(s) Installed.
                           [01]: Intel(R) Ethernet Connection (2) I219-V
                                 Connection Name: Ethernet
                                 DHCP Enabled:    Yes
                                 DHCP Server:     192.168.1.1
                                 IP address(es)
                                 [01]: 192.168.1.106
                                 [02]: fe80::718b:2274:3327:7b7f
                           [02]: VirtualBox Host-Only Ethernet Adapter
                                 Connection Name: VirtualBox Ethernet
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 192.168.56.1
                                 [02]: fe80::447b:e50a:c11e:22f2
                           [03]: Realtek 8822BE Wireless LAN 802.11ac PCI-E NIC
                                 Connection Name: WiFi
                                 Status:          Media disconnected
                           [04]: Bluetooth Device (Personal Area Network)
                                 Connection Name: Bluetooth Network Connection
                                 Status:          Media disconnected
                           [05]: Xbox Wireless Adapter for Windows
                                 Connection Name: Ethernet 2
                                 DHCP Enabled:    No
                                 IP address(es)
                           [06]: Hyper-V Virtual Ethernet Adapter
                                 Connection Name: vEthernet (Default Switch)
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 172.27.144.1
                                 [02]: fe80::599b:431b:833c:8ffc
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
```

### Steps to reproduce

  1. Setup cluster
  2. Suspend PC
  3. Resume PC
  4. Try to open crc console

### Expected

I would expect CRC to still be running after after resuming from the PC sleeping. Or at least to be able to start CRC again from the tray.

### Actual

Stopping and restarting doesn't help. I get this warning:

```
WARN Preflight checks failed during `crc start`, please try to run `crc setup` first in case you haven't done so yet
expected crcDaemon task to be in 'Running' but got 'Ready'
```

(running `crc setup` as suggested doesn't change anything. I have to delete the cluster first in order to get it reachable)

The solution is to run `crc delete -f; crc cleanup; crc setup; crc start`. This is a major loss in productivity! **This doesn't happen every time I suspend the machine**, but generally every time I come back to working with CRC after a few days of doing other work, I have to re-initialize it. All the members of my team are running openshift on Windows and have the same issue where they have to re-initialize every day or two.

CRC is often unreachable. I think the longest it's worked without a reset is probably 2 weeks, but generally it works for a day or two.

### Logs

Before gather the logs try following if that fix your issue
```bash
$ crc delete -f
$ crc cleanup
$ crc setup
$ crc start --log-level debug
```

Please consider posting the output of `crc start --log-level debug`  on http://gist.github.com/ and post the link in the issue.

```bash
# crc start --log-level debug
DEBU CRC version: 2.9.0+589ab2cd
DEBU OpenShift version: 4.11.3
DEBU Podman version: 4.2.0
DEBU Running 'crc start'
DEBU Total memory of system is 68639145984 bytes
WARN A new version (2.10.1) has been published on https://developers.redhat.com/content-gateway/file/pub/openshift-v4/clients/crc/2.10.1/crc-windows-installer.zip
DEBU Checking file: C:\Users\m\.crc\machines\crc\.crc-exist
DEBU Running 'Hyper-V\Get-VM crc | Select-Object -ExpandProperty State'
DEBU Checking file: C:\Users\m\.crc\machines\crc\.crc-exist
DEBU Copying 'C:\Users\m\.crc\cache\crc_hyperv_4.11.3_amd64\oc.exe' to 'C:\Users\m\.crc\bin\oc\oc.exe'
DEBU Copying 'C:\Users\m\.crc\cache\crc_hyperv_4.11.3_amd64\podman.exe' to 'C:\Users\m\.crc\bin\oc\podman.exe'
INFO Loading bundle: crc_hyperv_4.11.3_amd64...
DEBU Checking file: C:\Users\m\.crc\machines\crc\.crc-exist
DEBU Running 'Hyper-V\Get-VM crc | Select-Object -ExpandProperty State'
INFO A CRC VM for OpenShift 4.11.3 is already running
```

**Comment by gbraad** [comment]
On Monday I will try to follow-up on this and see what might be the problem. If by the end of the day you didn't see a reply, ping me

**Comment by gbraad** [comment]
I noticed you have VirtualBox installed. This might cause issues with the Hyper-V setup. \

> I would expect CRC to still be running after after resuming from the PC sleeping. 

> CRC is often unreachable. I think the longest it's worked without a reset is probably 2 weeks, but generally it works for a day or two.

Can you describe your usage. are you travelling a lot or closing the machine for 'sleep' (hibernate)? I do this daily, ...

When you notice it is unreachable, can you verifu if the VM is still running? Is the `crc daemon` still working? Is SSH still accessible?

```
PS> ssh 127.0.0.1 -p 2222
```

**Comment by mltucker** [comment]
Hey @gbraad, thanks a lot for checking this out!

Usage: this is a desktop PC (always wired internet). Generally I work with crc once per week, and use the PC a few more times in the week doing other work. I use Windows 10 "Sleep" when I'm not using it.

It's currently unreachable.

```bash
$ ssh 127.0.0.1 -p 2222
ssh: connect to host 127.0.0.1 port 2222: Connection refused
```

(though `crc start` still says `INFO A CRC VM for OpenShift 4.11.3 is already running`)

How can I check the status of the `crc daemon`? I suppose it wasn't running since I was able to run an instance of it and it didn't complain about opening `INFO listening \\.\pipe\crc-http` (whereas trying to run *another* daemon gave `daemon is already running`). However this didn't affect the behavior of `crc status` or `crc start`.

This also didn't affect the behavior of `crc stop; crc start`. I still see:

```
WARN Preflight checks failed during `crc start`, please try to run `crc setup` first in case you haven't done so yet
expected crcDaemon task to be in 'Running' but got 'Ready'
```

I am going to uninstall VirtualBox and completely reinstall crc. Though I'd be surprised if VirtualBox were a common denominator between me and my colleagues (who are all seeing this instability).

**Comment by gbraad** [comment]
> who are all seeing this instability

For some reason the daemon gets killed/stopped. I am not sure why, but can you check if the `~/.crc/` folder contains a log file?

**Comment by mltucker** [comment]
Hi @gbraad, sorry about the wait!

Here's [crc.log](https://github.com/crc-org/crc/files/10059055/crc.log)

And I had to truncate crcd.log (by removing all duplicate lines) since it was 350MB. [crcd_trunc.log](https://github.com/crc-org/crc/files/10059146/crcd_trunc.log). It's mostly filled by lines that look like this:
```
time="2022-11-21T18:36:32+01:00" level=debug msg="Checking file: C:\\Users\\m\\.crc\\machines\\crc\\.crc-exist"
time="2022-11-21T18:36:32+01:00" level=debug msg="Running 'Hyper-V\\Get-VM crc | Select-Object -ExpandProperty State'"
```

**Comment by cfergeau** [comment]
https://github.com/crc-org/crc/pull/3721 might help with this.

**Comment by anjannath** [comment]
@mltucker can you try out the latest release (`v2.33.0`) and see if this works for you now, we have added a fix for it in #3721
